solana/modules/wormhole-anchor-sdk/src/token_bridge/message.rs

```diff
@@ -33,26 +84,15 @@ impl AnchorDeserialize for TransferWithMeta {
 
         // Encoded amount should be the last 8 bytes of bytes 1 through 33,
         // otherwise we will have serious issues in the Token Bridge program.
-        let amount = {
-            let mut out = [0u8; 8];
-            out.copy_from_slice(&buf[25..33]);
-            u64::from_be_bytes(out)
-        };
-
-        let mut token_address = [0u8; 32];
-        token_address.copy_from_slice(&buf[33..65]);
+        let amount = U256::read(reader)?.as_u64();
 
-        let token_chain = to_u16_be(&buf[65..67]);
+        let token_address = Readable::read(reader)?;
+        let token_chain = Readable::read(reader)?;
+        let to_address = Readable::read(reader)?;
+        let to_chain = u16::read(reader)?;
+        let from_address = Readable::read(reader)?;
 
```

Comment:
So much better than the old code :heart:

Reply:
Yeah, definitely! It felt good rewriting that.